embeddings/config/lightning_config.py

```diff
@@ -0,0 +1,146 @@
+from dataclasses import dataclass, field
+from types import MappingProxyType
+from typing import Any, ClassVar, Dict, List, Optional, Set, Union
+
+from pytorch_lightning.accelerators import Accelerator
+
+from embeddings.config.base_config import AdvancedConfig, BasicConfig
+from embeddings.data.io import T_path
+from embeddings.utils.loggers import get_logger
+from embeddings.utils.utils import read_yaml
+
+T_kwarg = MappingProxyType[str, Any]
+
+_logger = get_logger(__name__)
+
+
+@dataclass
+class LightningConfigKeys:
+    PIPELINE_KEYS: ClassVar[Set[str]] = {
+        "batch_size",
+        "finetune_last_n_layers",
+        "embedding_name_or_path",
+    }
+    DATAMODULE_KEYS: ClassVar[Set[str]] = {"max_seq_length"}
+    TASK_MODEL_KEYS: ClassVar[Set[str]] = {
+        "learning_rate",
+        "optimizer",
+        "use_scheduler",
+        "warmup_steps",
+        "adam_epsilon",
+        "weight_decay",
+    }
+    TASK_TRAIN_KEYS: ClassVar[Set[str]] = {"max_epochs", "devices", "accelerator"}
+    MODEL_CONFIG_KEYS: ClassVar[Set[str]] = {"classifier_dropout"}
+    EARLY_STOPPING_KEYS: ClassVar[Set[str]] = {"monitor", "mode", "patience"}
+
+
+@dataclass
+class LightningConfigDefaultKwargs:
+    DEFAULT_DATAMODULE_KWARGS: ClassVar[T_kwarg] = MappingProxyType({"max_seq_length": None})
+    DEFAULT_TASK_MODEL_KWARGS: ClassVar[T_kwarg] = MappingProxyType({"use_scheduler": True})
+    DEFAULT_TASK_TRAIN_KWARGS: ClassVar[T_kwarg] = MappingProxyType(
+        {
+            "devices": "auto",
+            "accelerator": "auto",
+        }
+    )
+    DEFAULT_MODEL_CONFIG_KWARGS: ClassVar[T_kwarg] = MappingProxyType({"classifier_dropout": None})
+    DEFAULT_EARLY_STOPPING_KWARGS: ClassVar[T_kwarg] = MappingProxyType(
+        {
+            "monitor": "val/Loss",
+            "mode": "min",
+            "patience": 3,
+        }
+    )
+
+
+@dataclass
+class LightningBasicConfig(BasicConfig, LightningConfigKeys):
+    use_scheduler: bool = True
+    optimizer: str = "Adam"
+    warmup_steps: int = 100
+    learning_rate: float = 1e-4
+    adam_epsilon: float = 1e-8
+    weight_decay: float = 0.0
+    finetune_last_n_layers: int = -1
+    classifier_dropout: Optional[float] = None
+    max_seq_length: Optional[int] = None
+    mini_batch_size: int = 32
+    devices: Optional[Union[List[int], str, int]] = "auto"
+    max_epochs: Optional[int] = None
+    accelerator: Optional[Union[str, Accelerator]] = "auto"
+    monitor: str = "val/Loss"
```

Comment:
We need more descriptive name

Reply:
this is the name we pass to the callback so it has to stay the same but we can manage that, e.g., by setting the default values in different way, such as:

```
    monitor: str = LightningConfigDefaultKwargs.DEFAULT_EARLY_STOPPING_KWARGS["monitor"]
    mode: str = LightningConfigDefaultKwargs.DEFAULT_EARLY_STOPPING_KWARGS["mode"]
    patience: int = LightningConfigDefaultKwargs.DEFAULT_EARLY_STOPPING_KWARGS["patience"]
```

Reply:
Another solution

```
early_stopping_monitor_metric: str = "val/Loss"
early_stopping_mode: str
...
```
and then map this names to monitor, node, patience via some mapping function